backend/app/api/v1/reports.py

```
@@ -1,31 +1,112 @@
+async def make_report(
+    id_report: str = Query(...),
+    current_user: User = Depends(get_current_user),
+    db: AsyncSession = Depends(get_db),
+    ):
     try:
+        report = await report_service.get_report_by_id(db, id_report)
+        ensure_report_access(current_user, report)
+        
         html_path, pdf_path = await report_service.generate_report(db, id_report)
         return {"html_path": html_path, "pdf_path": pdf_path}
     except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)